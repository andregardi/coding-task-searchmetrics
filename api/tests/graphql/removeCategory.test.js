const { createTestClient } = require('apollo-server-testing');
const apolloServer = require('../../src/graphql/apolloServer');
const removeCategoryMutations = require('./__mutations__/removeCategory');

describe('removeCategory', () => {
  describe('on success', () => {
    it('it should match the snapshot', async () => {
      const { mutate } = createTestClient(apolloServer);
      const res = await mutate({
        mutation: removeCategoryMutations.successMutation,
      });

      expect(res).toMatchSnapshot();
    });
  });

  describe('if the category does not exist', () => {
    it('should return error', async () => {
      const { mutate } = createTestClient(apolloServer);
      const { errors } = await mutate({
        mutation: removeCategoryMutations.categoryDoesNotExist,
      });

      expect(errors[0].message).toMatch(
        /Category nonExistingCategory does not exist/
      );
    });
  });
});
