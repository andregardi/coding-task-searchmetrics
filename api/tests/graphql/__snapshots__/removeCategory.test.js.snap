// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeCategory on success it should match the snapshot 1`] = `
Object {
  "data": Object {
    "removeCategory": "cars",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
